src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java

```diff
@@ -93,10 +104,8 @@ private void createDownloadCloudSdkTask() {
 
               project.afterEvaluate(
                   p -> {
-                    downloadCloudSdkTask.setCloudSdkBuilderFactory(cloudSdkBuilderFactory);
-                    downloadCloudSdkTask.setManagedCloudSdkFactory(managedCloudSdkFactory);
-
                     if (toolsExtension.getCloudSdkHome() == null) {
+                      downloadCloudSdkTask.setManagedCloudSdk(managedCloudSdk);
```

Comment:
I'm starting to lean towards just setting this unconditionally (it might be null anyway) from above.